AdyenCashAppPay/CashAppPayComponent.swift

```diff
@@ -225,24 +231,37 @@ extension CashAppPayComponent: CashAppPayObserver {
         case let .approved(request, grants):
             submitApprovedRequest(with: grants, profile: request.customerProfile)
         case let .apiError(error):
-            fail(with: error)
+            fail(with: error, message: ErrorMessage.apiError)
         case let .networkError(error):
-            fail(with: error)
+            fail(with: error, message: error.localizedDescription)
         case let .unexpectedError(error):
-            fail(with: error)
+            fail(with: error, message: ErrorMessage.unexpectedError)
         case let .integrationError(error):
-            fail(with: error)
+            fail(with: error, message: ErrorMessage.integrationError)
         case .declined:
-            fail(with: Error.declined)
+            let error = Error.declined
+            fail(with: error, message: error.localizedDescription)
         default:
             break
         }
     }
 
-    private func fail(with error: Swift.Error) {
+    private func fail(with error: Swift.Error, message: String? = nil) {
         stopLoading()
+        sendThirdPartyErrorEvent(with: message)
```

Comment:
is this a backend decision? Otherwise we could try to prevent optionals
```suggestion
        sendThirdPartyErrorEvent(with: message ?? error.localizedDescription)
```

Reply:
we only send error description if it's manually created by us to make sure it doesn't contain sensitive data